### ISSUE TITLE
Repositories state refactor

### ISSUE THREAD

**Issue by lucasgarfield** [osbuild/image-builder-frontend#1924]
I'm getting a little worried the Wizard's state is becoming too confusing here - we've got a Payload Repositories field, a Custom Repositories field, and now a Recommended Repositories field.

I think that we should collapse these into a single field, SelectedRepositories, and then we can generate the payload and custom repositories fields when we generate the request.

This should make it much easier to ensure that bugs like this, where adding a new repo via the Proactive Assistance modal does not cause the repo to become selected on the repositories step, do not occur:
![image](https://github.com/osbuild/image-builder-frontend/assets/95542540/d3871c6e-1528-4b57-8eaa-9f81f059ed33)

**Comment by regexowl** [comment]
Fair enough, I'll gladly take this one. Originally the step was just migrated to V2 with minimal refactoring, but as I started working on repo recommendations I've noticed a few things I'd like to do differently.

Merging all the repos into one field in the state sounds good, the recommended ones can be differentiated by a flag. This way we shouldn't lose any functionality/detail needed to render things properly.
